fri/server/main.py

```diff
@@ -68,24 +68,36 @@ def build(dir):
     apikey = request.args.get('apikey') 
     dirname = secure_filename(dir) + "_" + apikey   
     makestudy_dir = dirname + "/" + graphml_file   #for makestudy
-    dir_path = os.path.abspath(os.path.join(concore_path, graphml_file)) #path for ./build
+    dir_path_pre = os.path.abspath(os.path.join(concore_path, graphml_file)) #path for ./build
+    dir_path = dir_path_pre + "_" + apikey
     if not os.path.exists(dir_path):
-        proc = call(["./makestudy", makestudy_dir], cwd=concore_path)
+        if(platform.uname()[0]=='Windows'):
+            proc= call(["makestudy", makestudy_dir], shell=True, cwd=concore_path)
+        else:
+            proc = call(["./makestudy", makestudy_dir], cwd=concore_path)
         if(proc == 0):
             resp = jsonify({'message': 'Directory successfully created'})
             resp.status_code = 201
         else:
             resp = jsonify({'message': 'There is an Error'})
-            resp.status_code = 500        
-    call(["./build"], cwd=dir_path)   
+            resp.status_code = 500   
+    if(platform.uname()[0]=='Windows'):
+        call(["build"], cwd=dir_path, shell=True)
+    else:
+        call(["./build"], cwd=dir_path)  
     return resp 
 
 
 @app.route('/debug/<dir>', methods=['POST'])
 def debug(dir):
     dir = secure_filename(dir)
-    dir_path = os.path.abspath(os.path.join(concore_path, dir))
-    proc = call(["./debug"], cwd=dir_path)
+    apikey = request.args.get('apikey')

```

Comment:
Please make sure the "apikey" is an optional argument.

In the presence of Kong configured to *expect* an API Key, it will drop the calls without a valid API key. So it will drop calls with apikey params that are empty or invalid (such as "xyz").

But when Kong is not configured (current status), we should support calls for both apikey that is not present or random alphanumeric characters (such as "xyz123"). 

Proposed Action: No additional checks are needed here at this point. But just make it an optional field.

Reply:
Unders

> Please make sure the "apikey" is an optional argument.
> 
> In the presence of Kong configured to _expect_ an API Key, it will drop the calls without a valid API key. So it will drop calls with apikey params that are empty or invalid (such as "xyz").
> 
> But when Kong is not configured (current status), we should support calls for both apikey that is not present or random alphanumeric characters (such as "xyz123").
> 
> Proposed Action: No additional checks are needed here at this point. But just make it an optional field.

Understood